src/algorithms.jl

```diff
@@ -54,19 +54,56 @@ function _show_alg(io::IO, alg::Algorithm)
 end
 
 @doc """
-    select_algorithm(f, A; kwargs...)
+    MatrixAlgebraKit.select_algorithm(f, A; kwargs...)
 
-Given some keyword arguments and an input `A`, decide on an algrithm to use for
+Given some keyword arguments and an input `A`, decide on an algorithm to use for
 implementing the function `f` on inputs of type `A`.
+
+In general, if an algorithm is specified explicitly through the `alg` keyword argument
+(either as an algorithm type, an algorithm name as a Symbol, or as an algorithm object),
+that algorithm will be used instead of selecting it automatically. However, that
+behavior may be modified for factorization functions and/or matrix types.
+
+In general, if the algorithm is not specified, a default algorithm specified by
+[`default_algorithm`](@ref) will be used.
 """
 function select_algorithm end
 
-function _select_algorithm(f, A::AbstractMatrix, alg::AbstractAlgorithm)
+function select_algorithm(f, A; alg=nothing, kwargs...)
```

Comment:
I am not sure using multiple dispatch via `_select_algorithm` to deal with the different cases of `alg` keywords is the right approach, as it might put more strain on the compiler. I don't think there is need to be able to extend this functionality, so something like
```julia
function select_algorithm(f, A; alg=nothing, kwargs...)
	if isnothing(alg)
		return default_algorithm(f, A; kwargs...)
	elseif alg isa AbstractAlgorithm
		isempty(kwargs) ||
			throw(ArgumentError("Additional keyword arguments are not allowed when an algorithm is specified."))
		return alg
	elseif
	...
	end
end
```
should work. Or does the `_select_algorithm` lowering actually help with the lowering?

Reply:
I'm totally fine with that, using dispatch was a bit arbitrary. I generally like using dispatch rather than if-statements purely as a style thing, but if you and the compiler prefer if-statements I can switch to that.

Reply:
If the `_select_algorithm` lowering is better for type stability, I am certainly fine with that, and there is no need to waste time trying to trick the compiler to make the `if elseif else` construction inferable as well.